examples/tvm/parser.py

```diff
@@ -0,0 +1,158 @@
+import pathlib
+import math
+from pprint import pprint
+import sys
+
+import libff as ff
+import libff.kv
+import libff.invoke
+
+import libff.kaas as kaas
+import numpy as np
+
+import json
+
+
+
+
+redisPwd = "Cd+OBWBEAXV0o2fg5yDrMjD9JUkW7J6MATWuGlRtkQXk/CBvf2HYEjKDYw4FC+eWPeVR8cQKWr7IztZy"
+redisPwd = "Cd+OBWBEAXV0o2fg5yDrMjD9JUkW7J6MATWuGlRtkQXk/CBvf2HYEjKDYw4FC+eWPeVR8cQKWr7IztZy"
+
+def getCtx(remote=False):
+        if remote:
+                objStore = ff.kv.Redis(pwd=redisPwd, serialize=True)
+        else:
+                objStore = ff.kv.Local(copyObjs=False, serialize=False)
+
+        return libff.invoke.RemoteCtx(None, objStore)
+
+
+
+
+
+
+''' Parse the given TVM graph in data, and return a kaas request and a list of buffers. '''
+def parse(inp, data, libffCtx, path, params, shapes):
+
+
+
+        nodes = data["nodes"]
+
+        kerns = []
+
+        buffers = []
+
+
+        param_counter = 0
+
+
+        copy_inp = np.copy(inp)
+        libffCtx.kv.put("0", copy_inp)
+        input_buffer = kaas.bufferSpec("0", copy_inp.nbytes)
+        buffers.append(input_buffer)    
+
+        
+        for i in range(1, len(nodes)):
+
+                output_size = np.array(data["attrs"]["shape"][1][i])
+                ty = data["attrs"]["dltype"][1][i]
+                node = nodes[i]
+
+
+                if node["op"] == "null":
+                    index = "p" + str(param_counter)
+                    param = params[index]
+                    
+                    zeros = np.zeros(output_size).astype(ty)
```

Comment:
what is this doing? Zeros and then copy the params in. Why not use the params directly?

Reply:
Thanks for the catch. That's an artifact of an older version where param had ctypes, so it wouldn't work with a lot of python functions. Hence, I needed to copy its data over into a strictly python array.